File: backend/app/schemas/dialogue.py
from pydantic import BaseModel
from typing import List, Dict

class DialogueResponse(BaseModel):
    dialogue_id: int
    situation: str
    conversation: List[Dict[str, str]]
    category: str
    lesson_id: int

class ChatRequest(BaseModel):
    dialogue_id: int
    conversation: List[Dict[str, str]]

class ChatResponse(BaseModel):
    dialogue_id: int
    conversation: List[Dict[str, str]]
    is_complete: bool

class TranslateResponse(BaseModel):
    translation: str

class SubmitDialogueRequest(BaseModel):
    dialogue_id: int
    conversation: List[Dict[str, str]]

class SubmitDialogueResponse(BaseModel):
    is_correct: bool
    feedback: str
    result_id: int
    dialogue_id: int

File: frontend/src/types/dialogue.ts
export interface Message {
  id: string;
  speaker: 'user' | 'AI';
  text: string;
}

export interface DialogueResponse {
  dialogue_id: number;
  situation: string;
  conversation: Message[];
  category: string;
  lesson_id: number;
}

export interface ChatRequest {
  dialogue_id: number;
  conversation: Message[];
}

export interface ChatResponse {
  dialogue_id: number;
  conversation: Message[];
  is_complete: boolean;
}

export interface TranslateResponse {
  translation: string;
}

export interface SubmitDialogueRequest {
  dialogue_id: number;
  conversation: Message[];
}

export interface SubmitDialogueResponse {
  is_correct: boolean;
  feedback: string;
  result_id?: number; // Optional, as it may not be used
  dialogue_id?: number; // Optional, as it may not be used
}